vyper/parser/stmt.py

```diff
@@ -203,41 +203,45 @@ def ann_assign(self):
+            if self.stmt.value is None:
+                raise StructureException(
+                    'Fresh variables must be initialized explicitly',
+                    self.stmt)
 
-                # Disallow assignment to None
-                if isinstance(sub.typ, NullType):
-                    raise InvalidLiteralException(
-                        (
-                            'Assignment to None is not allowed, use a default '
-                            'value or built-in `clear()`.'
-                        ),
-                        self.stmt
-                    )
+            sub = Expr(self.stmt.value, self.context).lll_node
 
-                is_valid_bytes32_assign = (
-                    isinstance(sub.typ, ByteArrayType) and sub.typ.maxlen == 32
-                ) and isinstance(typ, BaseType) and typ.typ == 'bytes32'
+            # Disallow assignment to None
+            if isinstance(sub.typ, NullType):
+                raise InvalidLiteralException(
+                    (
+                        'Assignment to None is not allowed, use a default '
+                        'value or built-in `clear()`.'
+                    ),
+                    self.stmt
+                )
 
-                # If bytes[32] to bytes32 assignment rewrite sub as bytes32.
-                if is_valid_bytes32_assign:
-                    sub = LLLnode(
-                        bytes_to_int(self.stmt.value.s),
-                        typ=BaseType('bytes32'),
-                        pos=getpos(self.stmt),
-                    )
+            is_valid_bytes32_assign = (
+                isinstance(sub.typ, ByteArrayType) and sub.typ.maxlen == 32
+            ) and isinstance(typ, BaseType) and typ.typ == 'bytes32'
 
-                self._check_valid_assign(sub)
-                self._check_same_variable_assign(sub)
-                variable_loc = LLLnode.from_list(
-                    pos,
-                    typ=typ,
-                    location='memory',
+            # If bytes[32] to bytes32 assignment rewrite sub as bytes32.
+            if is_valid_bytes32_assign:
+                sub = LLLnode(
+                    bytes_to_int(self.stmt.value.s),
+                    typ=BaseType('bytes32'),
                     pos=getpos(self.stmt),
                 )
-                o = make_setter(variable_loc, sub, 'memory', pos=getpos(self.stmt))
-                # o.pos = getpos(self.stmt) # TODO: Should this be here like in assign()?
+
+            self._check_valid_assign(sub)
+            self._check_same_variable_assign(sub)
+            variable_loc = LLLnode.from_list(
+                pos,
+                typ=typ,
+                location='memory',
+                pos=getpos(self.stmt),
+            )
+            o = make_setter(variable_loc, sub, 'memory', pos=getpos(self.stmt))
+            # o.pos = getpos(self.stmt) # TODO: Should this be here like in assign()?
```

Comment:
Wut dis?

Reply:
just carried over from the existing code (the indentation changed)

Reply:
Yeah, but do we know what it is and why it's there? Can it/should it be removed?

Reply:
It's exactly the stuff that #1609 will touch, i can remove the comment and see how load bearing it is during that job

Reply:
> Yeah, but do we know what it is and why it's there? Can it/should it be removed? If so, probably should do that separately.

yes, even if it wasn't a comment it would be idempotent with setting `pos=getpos(self.stmt)` in the constructor of `o`.